### ISSUE TITLE
Error updating the cluster while the bootstrap pod

### ISSUE THREAD

**Issue by AndersBennedsgaard** [Opster/opensearch-k8s-operator#501]
Since the operator uses a raw pod for the bootstrapper, any updates to the `OpenSearchCluster` resource or anything else which might trigger an update to the bootstrap pod, will result in the operator continuously failing with `Reconciler error [...] updating resource failed: Pod \"opensearch-bootstrap-0\" is invalid: spec: Forbidden: pod updates may not change fields other than [...]`.

Without having looked at the code, I believe that the fix is simple: upon triggering an update to the bootstrap pod, re-create it instead of trying to patch it.

**Comment by swoehrl-mw** [comment]
Hi @AndersBennedsgaard
As the bootstrap pod is only created for a few minutes during initial cluster startup this is probably not a really likely scenario. 

The problem is that the bootstrap pod does not have persistent storage, so if it is restarted while it is just connecting with other nodes to form a cluster, there might be problems. So this would need to be checked during implementation if it does not have unintended sideeffects.

**Comment by AndersBennedsgaard** [comment]
I agree that this isn't the most pressing issue in the world, but when bootstrapping a new cluster and you might have an error or two in your configuration, the operator spams this huge error (the entire pod spec is printed) very often. 
If pods can't be patched with other than a couple of very specific configurations (which would restart the application and might entroduce these errors you speak of) why even attempt to patch the pod in the first place?

**Comment by idanl21** [comment]
Hey @AndersBennedsgaard 
 Re-create it instead of trying to patch it should be a goo idea, it is uncommon scenario so we will try to work on it, but its not on top of our list.
 So we will update when we will fix that issue, but you are more then welcome to rise your own PR for that :)

**Comment by suckowbiz** [comment]
This bug avoids that you can apply coordinating only nodes. See: https://github.com/opensearch-project/opensearch-k8s-operator/issues/1023

**Comment by synhershko** [comment]
Operator should be in a much better state now, closing this as won't fix
